docs/Lance filter API vs regular pyarrow.txt

```diff
@@ -0,0 +1,6 @@
+The Lance project is a Rust-based library that aims to improve the performance and scalability of filtering operations in PyArrow. One of the key features of the Lance filter API is its ability to execute SQL-like WHERE clause expressions on Arrow tables, using the DataFusion engine under the hood.
```

Comment:
This is not quite correct. Lance is a disk-format and supports predicate pushdown with a similar interface as pyarrow dataset. The difference is that regular pyarrow datasets take Arrow compute expression whereas Lance accepts a SQL string.